internal/native/proxy.go

```diff
@@ -0,0 +1,679 @@
+	OnIndevEvent         func(event string)
+	OnRpcEvent           func(event string)
+	OnVideoStateChange   func(state VideoState)
+	OnNativeRestart      func()
+}
+
+func randomId(binaryLength int) string {
+	s := make([]byte, binaryLength)
+	_, err := rand.Read(s)
+	if err != nil {
+		nativeLogger.Error().Err(err).Msg("failed to generate random ID")
+		return strings.Repeat("0", binaryLength*2) // return all zeros if error
+	}
+	return hex.EncodeToString(s)
+}
+
+func (n *NativeOptions) toProxyOptions() *nativeProxyOptions {
+	// random 16 bytes hex string
+	handshakeMessage := randomId(16)
+	maxRestartAttempts := defaultMaxRestartAttempts
+	if n.MaxRestartAttempts > 0 {
+		maxRestartAttempts = n.MaxRestartAttempts
+	}
+	return &nativeProxyOptions{
+		SystemVersion:        n.SystemVersion,
+		AppVersion:           n.AppVersion,
+		DisplayRotation:      n.DisplayRotation,
+		DefaultQualityFactor: n.DefaultQualityFactor,
+		OnVideoFrameReceived: n.OnVideoFrameReceived,
+		OnIndevEvent:         n.OnIndevEvent,
+		OnRpcEvent:           n.OnRpcEvent,
+		OnVideoStateChange:   n.OnVideoStateChange,
+		OnNativeRestart:      n.OnNativeRestart,
+		HandshakeMessage:     handshakeMessage,
+		MaxRestartAttempts:   maxRestartAttempts,
+	}
+}
+
+func (p *nativeProxyOptions) toNativeOptions() *NativeOptions {
+	return &NativeOptions{
+		SystemVersion:        p.SystemVersion,
+		AppVersion:           p.AppVersion,
+		DisplayRotation:      p.DisplayRotation,
+		DefaultQualityFactor: p.DefaultQualityFactor,
+	}
+}
+
+// cmdWrapper wraps exec.Cmd to implement processCmd interface
+type cmdWrapper struct {
+	*exec.Cmd
+	stdoutHandler *nativeProxyStdoutHandler
+}
+
+func (c *cmdWrapper) GetProcess() interface {
+	Kill() error
+	Signal(sig interface{}) error
+} {
+	return &processWrapper{Process: c.Process}
+}
+
+type processWrapper struct {
+	*os.Process
+}
+
+func (p *processWrapper) Signal(sig interface{}) error {
+	if sig == nil {
+		// Check if process is alive by sending signal 0
+		return p.Process.Signal(os.Signal(syscall.Signal(0)))
+	}
+	if s, ok := sig.(os.Signal); ok {
+		return p.Process.Signal(s)
+	}
+	return fmt.Errorf("invalid signal type")
+}
+
+// NativeProxy is a proxy that communicates with a separate native process
+type NativeProxy struct {
+	nativeUnixSocket      string
+	videoStreamUnixSocket string
+	videoStreamListener   net.Listener
+	binaryPath            string
+
+	client      *GRPCClient
+	clientMu    sync.RWMutex
+	cmd         *cmdWrapper
+	logger      *zerolog.Logger
+	ready       chan struct{}
+	options     *nativeProxyOptions
+	restartMu   sync.Mutex
+	restarts    uint
+	stopped     bool
+	processWait chan error
+}
+
+// NewNativeProxy creates a new NativeProxy that spawns a separate process
+func NewNativeProxy(opts NativeOptions) (*NativeProxy, error) {
+	proxyOptions := opts.toProxyOptions()
+	proxyOptions.VideoStreamUnixSocket = fmt.Sprintf("@jetkvm/native/video-stream/%s", randomId(4))
```

Comment:
Would it be safer to use `os.CreateTemp()` to ensure we get a name that is known unused in an atomic action? Something like 
```golang
// Generate a unique socket path in /tmp with a prefix.
tmpFile, err := os.CreateTemp("/tmp", "jetkvm-native-video-stream-*.sock")
if err != nil {
    fmt.Printf("Failed to create temp file: %v\n", err)
    // pout os.Exit(1)
}
proxyOptions.VideoStreamUnixSocket := tmpFile.Name() 
tmpFile.Close()
os.Remove(socketPath) 
```